### PR TITLE
Multisite: Run activation hook on new site creation

### LINKED ISSUE #2929: Multisite: Run activation hook on new site creation
## Feature Description

Run activation hook on new site creation on a multisite.

[Example](https://github.com/spacedmonkey/super-simple-events/blob/master/public/class-super-simple-events.php#L332-L349) 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Hook into `wp_initialize_site` when roles are setup in new site.

## Testing Instructions

---

Fixes #2929

### CODE REVIEW

File: tests/phpunit/tests/Story_Post_Type.php [hunk 1]
```diff
@@ -289,4 +289,30 @@ public function test_add_caps_to_roles() {
...
+
+
+	/**
+	 * @covers ::add_caps_to_roles
+	 * @covers \Google\Web_Stories::new_site
+	 */
+	public function test_add_caps_to_roles_multisite() {
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'Test only runs on Multisite' );
+		}
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php]: Can't you just add `@group ms-required`? That does the same.

See https://github.com/WordPress/wordpress-develop/blob/022ebde54d6f06556382fa34f6e8d5fb0ee24d03/tests/phpunit/includes/abstract-testcase.php#L801-L803

Comment (spacedmonkey) [reply tests/phpunit/tests/Story_Post_Type.php]: This was copied from 

https://github.com/google/web-stories-wp/blob/7cf07181c4a39995df3552a4f50b86036c2b43af/tests/phpunit/tests/Updater.php#L175-L177
https://github.com/google/web-stories-wp/blob/7cf07181c4a39995df3552a4f50b86036c2b43af/tests/phpunit/tests/Uninstall.php#L76-L78

Comment (spacedmonkey) [reply tests/phpunit/tests/Story_Post_Type.php]: Should I change these as well?

Comment (swissspidy) [reply tests/phpunit/tests/Story_Post_Type.php]: Then let's fix it everywhere :-)

Comment (spacedmonkey) [reply tests/phpunit/tests/Story_Post_Type.php]: Fixed

File: tests/phpunit/tests/Story_Post_Type.php [hunk 2]
```diff
@@ -285,8 +285,31 @@ public function test_add_caps_to_roles() {
 		$editor        = get_role( 'editor' );
 
 		foreach ( $all_capabilities as $cap ) {
-			$administrator->has_cap( $cap );
-			$editor->has_cap( $cap );
+			$this->assertTrue( $administrator->has_cap( $cap ) );
+			$this->assertTrue( $editor->has_cap( $cap ) );
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php:289]: @spacedmonkey FYI, added some missing assertions here. Unfortunately didn't catch them before, but luckily the tests pass just fine still.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**